src/MacroTools/ObjectiveSystem/Objectives/UnitBased/ObjectiveAnyEnemyUnitInRects.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using MacroTools.Extensions;
+using MacroTools.FactionSystem;
+using MacroTools.QuestSystem;
+using WCSharp.Shared.Data;
+using static War3Api.Common;
+
+namespace MacroTools.ObjectiveSystem.Objectives.UnitBased
+{
+  /// <summary>
+  /// Completed when an eligible player moves a unit into the specified <see cref="Rectangle"/>.
+  /// </summary>
+  public sealed class ObjectiveAnyEnemyUnitInRects : Objective, IHasCompletingUnit
+  {
+    private readonly bool _heroOnly;
+    private readonly IEnumerable<Rectangle> _targetRects;
+    private readonly Faction? _faction;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ObjectiveAnyEnemyUnitInRects"/> class.
+    /// </summary>
+    /// <param name="targetRects">Where the player has to move a unit.</param>
+    /// <param name="faction"></param>
+    /// <param name="rectName">A user-friendly name for the area.</param>
+    /// <param name="heroOnly">If true, can only be completed with a hero.</param>
+    public ObjectiveAnyEnemyUnitInRects(IEnumerable<Rectangle> targetRects, Faction? faction, string rectName, bool heroOnly)
```

Comment:
I think we can just get rid of the heroOnly parameter; I can't imagine a quest where that would be relevant for this style of objective. Easy enough to add it back in later if a use becomes clear.